src/importexport/midi/internal/midishared/midifile.cpp

```diff
@@ -587,16 +586,15 @@ bool MidiFile::readEvent(MidiEvent* event)
             LOGD("readEvent: error 6");
             return false;
         }
-        data = new unsigned char[dataLen + 1];
         if (dataLen) {
-            read(data, dataLen);
+            data.resize(dataLen + 1);
```

Comment:
Seems the resize needs to be donw before the `if`, at least in my backport I do get a crash in the unit tests otherwise, `terminate called after throwing an instance of 'std::logic_error'`

Reply:
Seems the 2nd time this PR reveals an issue with the unit tests vs. those from Mu3